fe/fe-core/src/main/java/com/starrocks/planner/IcebergScanNode.java

```diff
@@ -0,0 +1,274 @@
+            }
+        }
+    }
+
+    @Override
+    public void finalize(Analyzer analyzer) throws UserException {
+        if (isFinalized) {
+            return;
+        }
+
+        LOG.debug("IcebergScanNode finalize. Tuple: {}", desc);
+        try {
+            getScanRangeLocations();
+        } catch (AnalysisException e) {
+            throw new UserException(e.getMessage());
+        }
+
+        // Min max tuple must be computed after analyzer.materializeSlots()
+        DescriptorTable descTbl = analyzer.getDescTbl();
+        minMaxTuple = descTbl.createTupleDescriptor();
+        PredicateUtils.computeMinMaxTupleAndConjuncts(analyzer, minMaxTuple, minMaxConjuncts, conjuncts);
+
+        computeStats(analyzer);
+        isFinalized = true;
+    }
+
+    @Override
+    public List<TScanRangeLocations> getScanRangeLocations(long maxScanRangeLength) {
+        return result;
+    }
+
+    public void getScanRangeLocations() throws UserException {
+        Optional<Snapshot> snapshot = IcebergUtil.getCurrentTableSnapshot(
+                srIcebergTable.getIcebergTable(), true);
+        if (!snapshot.isPresent()) {
+            LOG.info(String.format("Table %s has no snapshot!", srIcebergTable.getTable()));
+            return;
+        }
+        preProcessConjuncts();
+        for (FileScanTask task : IcebergUtil.getTableScan(
+                srIcebergTable.getIcebergTable(), snapshot.get(),
+                icebergPredicates, true).planFiles()) {
+            DataFile file = task.file();
+            LOG.info("Scan with file " + file.path() + ", file record count " + file.recordCount());
+
+            TScanRangeLocations scanRangeLocations = new TScanRangeLocations();
+
+            THdfsScanRange hdfsScanRange = new THdfsScanRange();
+            hdfsScanRange.setFull_path(file.path().toString());
+            hdfsScanRange.setOffset(task.start());
+            hdfsScanRange.setLength(task.length());
+            // For iceberg table we do not need partition id
+            hdfsScanRange.setPartition_id(-1);
+            hdfsScanRange.setFile_length(file.fileSizeInBytes());
+            hdfsScanRange.setFile_format(IcebergUtil.getHdfsFileFormat(file.format()).toThrift());
+            TScanRange scanRange = new TScanRange();
+            scanRange.setHdfs_scan_range(hdfsScanRange);
+            scanRangeLocations.setScan_range(scanRange);
+
+            // TODO: get hdfs block location information for scheduling, use iceberg meta cache
+            TScanRangeLocation scanRangeLocation = new TScanRangeLocation(new TNetworkAddress("-1", -1));
+            scanRangeLocations.addToLocations(scanRangeLocation);
+
+            result.add(scanRangeLocations);
+        }
+    }
+
+    @Override
+    protected String debugString() {
+        MoreObjects.ToStringHelper helper = MoreObjects.toStringHelper(this);
+        helper.addValue(super.debugString());
+        helper.addValue("icebergTable=" + srIcebergTable.getName());
```

Comment:
We can just use `helper.add("icebergTable", srIcebergTable)` because the helper will concat the key&value automatically.